crates/uwheel/src/window/hammer_slide.rs

```diff
@@ -0,0 +1,357 @@
+use crate::{window::circular_queue::CircularQueue, Aggregator};
+use std::{marker::PhantomData, vec::Vec};
+
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "serde", serde(bound = "A: Default"))]
+#[derive(Default)]
+pub struct HammerSlide<A: Aggregator> {
+    m_window_slide: usize,
+
+    m_capacity: usize,
+    m_istack_size: usize,
+    m_istack_ptr: isize,
+    m_ostack_size: usize,
+    m_ostack_ptr: isize,
+    m_istack_val: A::PartialAggregate, // should be PartialAggregate
+
+    m_queue: CircularQueue<A>, // should be any possible input type
+    m_ostack_val: Vec<A::PartialAggregate>,
+
+    _phantom: PhantomData<(A::Input, A::Aggregate)>,
+}
+
+pub const DEFAULT_WINDOW_SLIDE: usize = 1;
+
+impl<A: Aggregator> HammerSlide<A> {
+    pub fn new(window_size: usize, window_slide: usize) -> Self {
```

Comment:
Does HammerSlide need both window size and slide? Is the number of slices/pairs (capacity) not enough?